### ISSUE TITLE
Can't read full text of .pdf file

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/runtime#59998]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Gave-invalid-input-on-my-previous-report/1541811)._

---
[severity:I'm unable to use this version]
Just a few minutes ago I reported that File.ReadAllText(filename) didn't return the whole of the file when the filename was a .pdf file, stopping at the first "%%EOF" string. I then said that when I copied the file to a .txt extension, it read in the entire file. My mistake. It's still stopping at the first "%%EOF" string.

My apologies. But it's still a bug.

---
#### Feedback Bot on 9/30/2021, 04:45 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Natalia Kondratyeva on 10/5/2021, 04:28 PM: 

<p>Hello Larry,<br>
Thank you for the report.</p>
<p>For convenience, let me cite the information from your previous ticket – <a target='_blank' href="https://developercommunity.visualstudio.com/t/Cant-read-full-text-of-pdf-file/1541806">https://developercommunity.visualstudio.com/t/Cant-read-full-text-of-pdf-file/1541806</a></p>
<blockquote>
<p>[severity:I’m unable to use this version]<br>
I’m trying to learn about the internal format of a .pdf file. I wrote<br>
string pdf = File.ReadAllText(filename);<br>
Instead of returning a string of length 82065, it returned only the first 78500 chars. It stopped at the first “%%EOF” string, which isn’t right.<br>
Copied the .pdf file unchanged to a .txt file and got all 82065 chars.</p>
<p>VS 2022, version 4.0, .Net 6.0.100-rc.1.21458.32</p>
</blockquote>
<p>As it concerns .NET 6, I will now move this ticket to <a target='_blank' href="https://github.com/dotnet/runtime">dotnet/runtime</a> GitHub repo.</p>

---
### Original Solutions
(no solutions)

**Comment by samsosa** [comment]
**tl;dr**
Use
```cs 
File.ReadAllText(pdfFile, Encoding.ASCII)
```
or
```cs 
File.ReadAllText(pdfFile, Encoding.Latin1)
```
because
```cs 
File.ReadAllText(pdfFile)
```
uses a multi-byte utf8 encoding.



**This is not a bug:**
This is the result of the "weird" way of reading a non-text file. The reason why the lengths are not equal is because of the encoding, which is UTF8 when File.ReadAllText() is used without parameters. UTF8 is multi byte and in the binary data stream there are sometimes character sequences which consist of more than one byte. But in C# string.Length counts only "single" characters and not the multi bytes which are actual needed for byte encoding.

**Comment by carlossanlop** [comment]
Thanks for your help, @samsosa.

@vsfeedback I have no way of pinging Larry Smith, who reported this issue on Developer Community. I can't comment on the original issue either (it's read-only now). So if you can let him know of the above suggestion to fix his issue, that would be great.
